File: backend/app/schemas/public.py
from pydantic import BaseModel, EmailStr, Field


class ContatoForm(BaseModel):
    nome: str = Field(min_length=1, max_length=200)
    email: EmailStr
    escritorio: str | None = Field(default=None, max_length=200)
    assunto: str | None = Field(default=None, max_length=200)
    mensagem: str = Field(min_length=1, max_length=4000)

File: frontend/src/api/public.ts
import { api } from "./client";

export interface ContatoForm {
  nome: string;
  email: string;
  escritorio?: string;
  assunto?: string;
  mensagem: string;
}

export const publicApi = {
  enviarContato: (payload: ContatoForm) =>
    api.post<{ enviado: boolean }>("/public/contato", payload),
};
